src/AutoRest.CSharp/Common/Output/Builders/SerializationBuilder.cs

```diff
@@ -417,14 +421,30 @@ private IEnumerable<JsonPropertySerialization> GetPropertySerializationsFromBag(
         public JsonObjectSerialization BuildJsonObjectSerialization(ObjectSchema objectSchema, SchemaObjectType objectType)
         {
             var propertyBag = new SerializationPropertyBag();
-            foreach (var objectTypeLevel in objectType.EnumerateHierarchy())
+            var addToPropertyBag = (ObjectType curObj, ObjectTypeProperty curProp) =>
             {
-                foreach (var objectTypeProperty in objectTypeLevel.Properties)
+                if (curProp.SchemaProperty != null && !propertyBag.Properties.ContainsKey(curProp))
                 {
-                    if (objectTypeProperty.SchemaProperty != null)
+                    var ms = objectType.GetForMemberSerialization(curProp.Declaration.Name);
+                    if (ms == null && curObj is SchemaObjectType sot)
                     {
-                        propertyBag.Properties.Add(objectTypeProperty, objectType.GetForMemberSerialization(objectTypeProperty.Declaration.Name));
+                        ms = sot.GetForMemberSerialization(curProp.Declaration.Name);
                     }
+                    propertyBag.Properties.Add(curProp, ms);
+                }
+            };
```

Comment:
Instead of a lambda, we should define a local method here according to the scenario here. Such as
```suggestion
            static void AddToPropertyBag(PropertyBag propertyBag, ObjectType curObj, ObjectTypeProperty curProp)
            {
                if (curProp.SchemaProperty != null && !propertyBag.Properties.ContainsKey(curProp))
                {
                    var ms = objectType.GetForMemberSerialization(curProp.Declaration.Name);
                    if (ms == null && curObj is SchemaObjectType sot)
                    {
                        ms = sot.GetForMemberSerialization(curProp.Declaration.Name);
                    }
                    propertyBag.Properties.Add(curProp, ms);
                }
            };
```

Reply:
use lamda can further limit the impact scope of the method. but anyway, make it a private method to make it the same style of the rest code.